packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -0,0 +1,278 @@
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/continuous_scan_model.dart';
+import 'package:smooth_app/main.dart';
+import 'package:smooth_app/pages/scan/scan_page_helper.dart'
+    as scan_page_helper;
+import 'package:smooth_app/widgets/smooth_product_carousel.dart';
+import 'package:smooth_ui_library/animations/smooth_reveal_animation.dart';
+import 'package:smooth_ui_library/widgets/smooth_view_finder.dart';
+import 'package:visibility_detector/visibility_detector.dart';
+
+class MLKitScannerPage extends StatefulWidget {
+  const MLKitScannerPage({Key? key}) : super(key: key);
+
+  @override
+  MLKitScannerPageState createState() => MLKitScannerPageState();
+}
+
+class MLKitScannerPageState extends State<MLKitScannerPage> {
+  BarcodeScanner? barcodeScanner = GoogleMlKit.vision.barcodeScanner();
+  late ContinuousScanModel _model;
+  CameraController? _controller;
+  int _cameraIndex = 0;
+  CameraLensDirection cameraLensDirection = CameraLensDirection.back;
+  bool isBusy = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (cameras.any(
+      (CameraDescription element) =>
+          element.lensDirection == cameraLensDirection &&
+          element.sensorOrientation == 90,
+    )) {
+      _cameraIndex = cameras.indexOf(
+        cameras.firstWhere((CameraDescription element) =>
+            element.lensDirection == cameraLensDirection &&
+            element.sensorOrientation == 90),
+      );
+    } else if (cameras.any((CameraDescription element) =>
+        element.lensDirection == cameraLensDirection)) {
+      _cameraIndex = cameras.indexOf(
+        cameras.firstWhere(
+          (CameraDescription element) =>
+              element.lensDirection == cameraLensDirection,
+        ),
+      );
+    }
+
+    _startLiveFeed();
+  }
+
+  @override
+  void dispose() {
+    _stopLiveFeed();
+    _disposeLiveFeed();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: VisibilityDetector(
+        key: const Key('VisibilityDetector ML Kit'),
+        onVisibilityChanged: (VisibilityInfo visibilityInfo) {
+          if (visibilityInfo.visibleFraction == 0.0) {
+            _stopLiveFeed();
+          } else {
+            _startLiveFeed();
+          }
+        },
+        child: _liveFeedBody(),
+      ),
+    );
+  }
+
+  Widget _liveFeedBody() {
+    if (_controller?.value.isInitialized == false) {
+      return Container();
+    }
+
+    final Size size = MediaQuery.of(context).size;
+    // calculate scale depending on screen and camera ratios
+    // this is actually size.aspectRatio / (1 / camera.aspectRatio)
+    // because camera preview size is received as landscape
+    // but we're calculating for portrait orientation
+    double scale = size.aspectRatio * _controller!.value.aspectRatio;
+
+    // to prevent scaling down, invert the value
+    if (scale < 1) {
+      scale = 1 / scale;
+    }
```

Comment:
This calculation is from here: 

https://stackoverflow.com/a/61487358/13313941

without that the camera looks really streched when moving around

Reply:
If so please put that in the code's comment, not here.

Reply:
Fair enough